src/otp/otp.service.spec.ts

```diff
@@ -66,4 +66,43 @@ describe("OtpService", () => {
 
     expect(actualResult).toBe(false);
   });
+
+  it("OTP should be created for email sending", async () => {
+    const mockOtp = MockFactory(OtpFixture).one().withEmailChannel();
+
+    jest
+      .spyOn(otpRepository, "findOne")
+      .mockResolvedValue(Promise.resolve(null));
+
+    jest
+      .spyOn(otpRepository, "save")
+      .mockResolvedValue(Promise.resolve(mockOtp));
+
+    const actualResult = await otpService.createEmailOtpAsync(
+      mockOtp.channel.email
+    );
+
+    expect(actualResult.otpValue).toBe(mockOtp.value);
+    expect(actualResult.isSent).toBe(true);
+    expect(actualResult.expiresAt).toBe(mockOtp.expiresAt);
+  });
+
+  it("new OTP should not be created to send email", async () => {
+    const mockOtp = MockFactory(OtpFixture).one().withEmailChannel();
+
+    jest
+      .spyOn(otpRepository, "findOne")
+      .mockResolvedValue(Promise.resolve(mockOtp));
+
+    jest.spyOn(otpRepository, "save").mockResolvedValue(Promise.resolve(null));
```

Comment:
You don't need this spy, since if an unexpired otp is already exists, you won't save anything.